src/aotcompile.cpp

```diff
@@ -297,8 +297,10 @@ void *jl_create_native_impl(jl_array_t *methods, LLVMOrcThreadSafeModuleRef llvm
         params.world = compile_for[worlds];
         if (!params.world)
             continue;
-        // Don't emit methods for the typeinf_world with extern policy
-        if (policy != CompilationPolicy::Default && params.world == jl_typeinf_world)
+        // Don't emit methods for the typeinf_world with extern policy, or if stripping IR
+        // (since we won't be able to use the compiler then anyway)
```

Comment:
Call the compiler is part of the spec for `broadcast` though, among others?

Reply:
True, but somehow this works in testing --- is it possible the code for the current world is applicable to the typeinf world as well? Otherwise I'm not sure how it's working...

Reply:
Most of it probably is